Revert "fix incorrect mimetype inferrence"

cketti opened the pull request:
This reverts commit 2264fffe02b3f4e9e4425a57667b9b49053a1a12.

See https://github.com/thunderbird/thunderbird-android/pull/8048#issuecomment-2304729673

kewisch reviewed (approved):
I wonder if it would make sense to infer mime type by magic numbers instead. This might give us the best of both worlds without making assumptions about text/plain specifically.

cketti commented:
That would create the same problem. It doesn't help the ecosystem if things work with our client because we implemented some non-standard magic, but fails with other clients because they don't.

Software that generates emails that use `text/plain` for an attachment that isn't plain text is broken and needs to be fixed. When we don't work around such a bug, chances are higher the bug will be discovered and fixed sooner rather than later.
